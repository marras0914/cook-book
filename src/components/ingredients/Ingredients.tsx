import React, { SFC } from 'react';
import { Wrapper, StyledList, StyledListItem } from '../../common/styles';
import { BackButton } from '../../common';
import { Ingredient } from '../../types';

interface IngredientsProps {
  ingredients: Ingredient[];
}

const Ingredients: SFC<IngredientsProps> = ({ ingredients }) => {
  return (
    <Wrapper>
      <StyledList>
        {ingredients.map(item => (
          <StyledListItem key={item.name}>
            <div>{item.quantity}</div>
            <div>{item.name}</div>
          </StyledListItem>
        ))}
      </StyledList>
      <BackButton />
    </Wrapper>
  );
};

export default Ingredients;
