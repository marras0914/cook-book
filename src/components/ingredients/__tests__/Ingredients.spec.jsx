import React from 'react';
import { shallow } from 'enzyme';
import Ingredients from '../Ingredients';

describe('Ingredients', () => {
  const recipe = [
    {
      name: 'poblano pepper',
      quantity: '3',
    },
    {
      name: 'chicken',
      quantity: '3 lbs',
    },
  ];

  it('Renders a list of ingredients', () => {
    const ingredients = shallow(<Ingredients ingredients={recipe} />);
    expect(ingredients).toMatchSnapshot();
  });

  // it('Renders link to Google with classname', () => {
  //   const link = shallow(
  //     <Link href="http://google.com" className="my-link-class">
  //       {' '}
  //       Link to Google
  //     </Link>,
  //   );
  //   expect(link).toMatchSnapshot();
  // });
});
