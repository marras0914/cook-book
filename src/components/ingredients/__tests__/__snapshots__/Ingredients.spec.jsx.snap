// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredients Renders a list of ingredients 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      key="poblano pepper"
    >
      <div>
        3
      </div>
      <div>
        poblano pepper
      </div>
    </styled.div>
    <styled.div
      key="chicken"
    >
      <div>
        3 lbs
      </div>
      <div>
        chicken
      </div>
    </styled.div>
  </styled.div>
  <BackButton />
</styled.div>
`;
